resolwe/flow/tests/test_purge.py

```diff
@@ -0,0 +1,223 @@
+                        'type': 'basic:dir:'
+                    },
+                    {
+                        'name': 'sample_file_list',
+                        'label': 'Sample list of output files',
+                        'type': 'list:basic:file:'
+                    },
+                    {
+                        'name': 'sample_dir_list',
+                        'label': 'Sample list of output directories',
+                        'type': 'list:basic:dir:'
+                    }
+                ],
+                run={
+                    'bash': """
+touch these files should be removed
+touch this_file_should_stay
+mkdir -p directory/should/be
+touch directory/should/be/removed
+touch directory/should/be/removed2
+mkdir -p stay/directory
+touch stay/directory/file1
+touch stay/directory/file2
+touch entry1 entry2 entry3
+mkdir dir1 dir2 dir3
+touch dir1/a dir2/b dir3/c dir3/d
+touch ref1 ref2 ref3 ref4
+mkdir refs
+touch refs/a refs/b
+
+re-save-file sample_file this_file_should_stay ref3
+re-save-dir sample_dir stay ref4
+re-save-file-list sample_file_list entry1 entry2:ref1 entry3:refs
+re-save-dir-list sample_dir_list dir1:ref2 dir2 dir3
+""",
+                }
+            )
+        )
+
+        self.assertFilesRemoved(data, 'these', 'files', 'should', 'be', 'removed', 'directory')
+        self.assertFilesNotRemoved(data, 'this_file_should_stay', 'stay/directory/file1', 'stay/directory/file2',
+                                   'entry1', 'entry2', 'entry3', 'dir1/a', 'dir2/b', 'dir3/c', 'dir3/d', 'ref1',
+                                   'ref2', 'ref3', 'ref4', 'refs/a', 'refs/b')
+
+    def assertFieldWorks(self, field_type, field_value, script_setup, script_save, removed, not_removed):
+        field_schema = {
+            'name': 'sample',
+            'label': 'Sample output',
+            'type': field_type
+        }
+
+        # Test simulated operation.
+        simulated_root = tempfile.mkdtemp()
+        try:
+            for filename in removed + not_removed:
+                directory, basename = os.path.split(filename)
+                if directory:
+                    try:
+                        os.makedirs(os.path.join(simulated_root, directory))
+                    except OSError:
+                        pass
+
+                if basename:
+                    with open(os.path.join(simulated_root, filename), 'w'):
+                        pass
+
+            unreferenced_files = purge.get_purge_files(
+                simulated_root,
+                output={'sample': field_value},
+                output_schema=[field_schema],
+                descriptor={},
+                descriptor_schema=[]
+            )
+
+            for filename in not_removed:
+                self.assertNotIn(os.path.realpath(os.path.join(simulated_root, filename)), unreferenced_files)
+
+            for filename in removed:
+                filename = os.path.realpath(os.path.join(simulated_root, filename))
```

Comment:
Why do you use `realpath`? It changes the path (`/var/folders/6q/x6bxkpk51xz37hkr06crmk9h0000gn/T/tmpGvPbS2/remove_this/` -> `/private/var/folders/6q/x6bxkpk51xz37hkr06crmk9h0000gn/T/tmpGvPbS2/remove_this`) and causes tests to fail on OS X. Without it tests fail because trailing slashes, but paths are correct.